### PR TITLE
fix(maillist): Ensure actions involving 2 folders update counts on both

### PR DESCRIPTION
**Pull request by castaway**
Fixes issues with displayed folder counts being incorrect after message moves (including to spam/trash) - intermittent issue that happens when a move is made when an index sync is also in progress

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -917,6 +921,10 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
   dropToFolder(folderId): void {
     const messageIds = this.canvastable.rows.selectedMessageIds();
 
+    let currentFolderId;
+    this.messagelistservice.folderListSubject.subscribe(folders => {
```
Comment (tadzik) [inline src/app/app.component.ts:925]: I think a race is possible here: messageActionsHandler should probably wait for this call to finish before it fires its own.

**Review by tadzik — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 1 above)
Comment (castaway) [reply src/app/app.component.ts:925]: folderListSubject is a BehavorSubject, which is documented as returning its current value immediately..
we already use the exact same construction in existing code ... so I think its fine?

Comment (castaway) [reply src/app/app.component.ts]: Moved the variable closer to its use, and switched to behaviorsubject .value, which we don't have to wait on. (If user's are running actions against messages/folders, and don't have folders, something is quite broken..)

File: src/app/app.component.ts [hunk 2]
```diff
@@ -963,8 +975,12 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
             this.clearSelection();
             this.canvastable.rows.clearOpenedRow();
           },
-          updateRemote: (msgIds: number[]) =>
-            this.messagelistservice.rmmapi.moveToFolder(msgIds, folder)
+          updateRemote: (msgIds: number[]) => {
+            console.log('update remote: start');
```
[the remark this answers is not in the document]
Comment (castaway) [reply src/app/app.component.ts]: fixed

**Review by tadzik — APPROVED** [review]
Looks good ~~– assuming that the logs are staying intentionally~~ :)

**Review by hernan604 — APPROVED** [review]

**Pull request merged**